File: files/CH-128.py
import itertools
import pandas as pd

path = "CH-128 Cartesian Product.xlsx"
test = pd.read_excel(path, usecols = "C")
lets = ["A","B","C","D"]
combs = [''.join(p) for p in itertools.product(lets, repeat=3)]

print(all(combs == test["Result"])) # True

File: files/100-199/128/CH-128.R
library(tidyverse)
library(readxl)

path = "files/CH-128 Cartesian Product.xlsx"
test  = read_excel(path, range = "C1:C65")

lets = c("A","B","C","D")

result = expand.grid(lets,lets, lets) %>%
  unite("result", Var1:Var3, sep = "") %>%
  arrange(result)

all.equal(result$result, test$Result)
#> [1] TRUE